report/data_tasks/report/create_from_scratch/99_hubspot_write/02_contact_property_update/01_create_view.sql

```diff
@@ -2,18 +2,68 @@ DROP VIEW IF EXISTS hubspot.contact_property_update CASCADE;
+        -- We are going to assume people never _stop_ being a teacher. This means
+        -- we will never remove the `Teacher` status from a contact. The upside is
+        -- we will update far fewer records, which should happen quicker.
+        JOIN lms.users ON
+            -- Emails in the contacts table are already stored as lower case
+            contacts.email = LOWER(users.email)
+            -- We definitely want to NOT join onto student information to keep that
+            -- out of Hubspot
+            AND users.is_teacher = True
+        GROUP BY contacts.id
+    )
+
     SELECT
-        contacts.id AS hs_object_id,
-        'Teacher' AS lms_role
-    FROM hubspot.contacts
-    -- We are going to assume people never _stop_ being a teacher. This means
-    -- we will never remove the `Teacher` status from a contact. The upside is
-    -- we will update far fewer records, which should happen quicker.
-    JOIN lms.users ON
-        -- Emails in the contacts table are already stored as lower case
-        contacts.email = LOWER(users.email)
-        -- We definitely want to NOT join onto student information to keep that
-        -- out of Hubspot
-        AND users.is_teacher = True
-    GROUP BY contacts.id
+        contact_users.hs_object_id,
+        contact_users.h_user_id,
+        'Teacher' AS lms_role,
+        contact_users.lms_first_login,
+        last_login.lms_last_login,
+        CURRENT_DATE AS reporting_last_update,
+        COALESCE(course_activity.lms_annotations_this_semester, 0) AS lms_annotations_this_semester,
+        COALESCE(course_activity.lms_active_courses_this_semester, 0) AS lms_active_courses_this_semester
+    FROM contact_users
+
+    LEFT OUTER JOIN (
+        SELECT
+            user_id,
+            SUM(annotation_count) AS lms_annotations_this_semester,
+            COUNT(DISTINCT(group_roles.group_id)) AS lms_active_courses_this_semester
+        FROM lms.users
+        JOIN lms.group_roles ON
+            group_roles.user_id = users.id
+            AND group_roles.role = 'teacher'
+        JOIN lms.group_bubbled_activity ON
+            group_bubbled_activity.group_id = group_roles.group_id
+            AND group_bubbled_activity.created_week >= report.multi_truncate('semester', CURRENT_DATE)
+        JOIN lms.groups ON
+            -- We are limiting by courses, because we want to count courses. We
+            -- still get the right counts for non-unique items by using the
+            -- bubbled version of the table
+            groups.id = group_roles.group_id
+            AND groups.group_type = 'course'
+        GROUP BY user_id
+    ) AS course_activity ON
+        course_activity.user_id = contact_users.h_user_id
+
+    LEFT OUTER JOIN (
+        SELECT
+            user_id,
+            MAX(timestamp_week) AS lms_last_login
+        FROM lms.events
+        WHERE
+            user_id IS NOT NULL
+            AND event_type IN ('configured_launch', 'deep_linking')
```

Comment:
These are both types of login. I don't think it's safe to assume all future events will be